### PR TITLE
Expert flow terms

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- visit `/new/?dd=cs:~cmarstech/mattermost`
- scroll down and enter budget
- scroll down and you should see a checkbox to agree to terms
- click the link to view the terms
- a popup should appear with the terms

**Comment by huwshimi** [comment]
[Add agree to terms to the expert flow](https://trello.com/c/GlvnkHfp/653-add-agree-to-terms-to-the-expert-flow)

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8692/

File: jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
+/* Copyright (C) 2017 Canonical Ltd. */
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js]: 2018

File: jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+'use strict';
+
+const PropTypes = require('prop-types');
+const React = require('react');
+const classNames = require('classnames');
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js]: Put those in alphabetical order?

File: jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+const GenericButton = require('../../generic-button/generic-button');
+const TermsPopup = require('../../terms-popup/terms-popup');
+
+class DeploymentAgreements extends React.Component {
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js]: Please add a docstring for the class.

File: jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      showTerms: false
+    };
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js]: We could bind _toggleTerms here and avoid doing it every time below? [shrug]

File: jujugui/static/gui/src/app/components/deployment-flow/agreements/test-agreements.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+'use strict';
+
+const React = require('react');
+const enzyme = require('enzyme');
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/deployment-flow/agreements/test-agreements.js]: Alphabetical order.

File: jujugui/static/gui/src/app/components/deployment-flow/agreements/test-agreements.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
+/* Copyright (C) 2017 Canonical Ltd. */
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/deployment-flow/agreements/test-agreements.js]: 2018

**Comment by frankban** [comment]
QA needed to be done with the pay and experts flags against staging.
QA is OK, I was able to reed Let's Encrypt agreement.

File: jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js (hunk 1 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js]: Oops, done.

File: jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js (hunk 2 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js]: Done.

File: jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js (hunk 3 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js]: Done.

File: jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js (hunk 4 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/components/deployment-flow/agreements/agreements.js]: I think we looked at that pattern when we switched over to classes, but it makes it slightly harder to debug as you have to remember to check that functions have been bound and can result in some strange errors. Happy to revisit if people are keen.

File: jujugui/static/gui/src/app/components/deployment-flow/agreements/test-agreements.js (hunk 5 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/components/deployment-flow/agreements/test-agreements.js]: Done

File: jujugui/static/gui/src/app/components/deployment-flow/agreements/test-agreements.js (hunk 6 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/components/deployment-flow/agreements/test-agreements.js]: Done.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8694/

**Pull request merged**